web/vtadmin/src/components/Icon.tsx

```diff
@@ -0,0 +1,82 @@
+import * as icons from '../icons';
+
+interface Props {
+    className?: string;
+    icon: Icons;
+}
+
+// All icons are from the VTAdmin Figma icon library:
+// https://www.figma.com/file/By3SoETBRHpOirv3Ctfxdq/Designs
+export const Icon = ({ icon, ...props }: Props) => {
```

Comment:
Between the SVGs and the `Icons` enum, I'm... _happy_ with this implementation. It's one of the best SVG/icon implementations I've worked with, but it's still pretty clunky. 😕 The trade-off here is verbosity for type-safety. 

It's possible we could reduce almost all of this boilerplate through some cheeky codegen, but that's particularly well-supported in TypeScript (compared to Go) so it'll take some exploration.